index.js

```diff
@@ -74,25 +75,29 @@ function compression (options) {
 
     // proxy
 
-    res.write = function write (chunk, encoding) {
+    res.write = function write (chunk, encoding, cb) {
       if (ended) {
         return false
       }
 
+      cb = (res._write.length === 3) ? cb : null
```

Comment:
You make an excellent point. I was attempting to

> Detect through some mechanism if the upstream stream supports callback arguments

but you're right--this won't work. As I was looking for other options, I came across `res._writeRaw` which has 3 parameters when there's a callback available.

I know that this is just like evaluating the original `res.write`, but I don't know of another way to check whether something supports callbacks than to actually check the argument list of some function. Since this is a little more internal, I thought it might be less likely to be overwritten, but it's no home run.

I looked for other options too, but didn't find anything. For example, `res._writableState.writecb` exists in node 4 so I checked node 0.10, but it exists there as well.

Any input on how to determine whether the upstream stream supports callbacks in a maintainable way?